compiler/src/iree/compiler/ConstEval/JitGlobals.cpp

```diff
@@ -212,6 +260,58 @@ class ProgramBuilder {
     return m;
   }
 
+  // Returns true if the given |initializerOp| is a constant expression that is
+  // able to be evaluated by this pass.
+  bool isConstExpr(IREE::Util::InitializerOp initializerOp) {
+    // Bail if multiple blocks. It would be possible to support these in theory
+    // but unclear if worth it in practice given the predominance of SCF at the
+    // levels we run things. What we'd require is adding a single exit block
+    // that stored to the globals unconditionally.
+    if (!initializerOp.getBody().hasOneBlock()) {
+      emitWarning(initializerOp.getLoc())
+          << "skipping consteval initializer: initializers with >1 block not "
+             "yet supported";
+      return false;
+    }
+
+    // Don't allow anything with parameters today. We could handle these by
+    // passing in the parameter file for use but would likely also want to bind
+    // a writeable parameter file to produce into.
+    if (isParameterized(sourceSymbolTable, initializerOp)) {
+      emitWarning(initializerOp.getLoc())
+          << "skipping consteval initializer: uses parameters";
+      return false;
+    }
+
+    // Today we require that all values are constant expressions. We could slice
+    // out just the ones that are.
+    for (auto &op : initializerOp.getOps()) {
+      // DO NOT SUBMIT
+      op.dump();
```

Comment:
DNS

Reply:
wanted to be able to see what ops were processed on the bots - will remove